src/ML-POD/fitpod_command.cpp

```diff
@@ -0,0 +1,1554 @@
+  memory->create(data.energy, n, "fitpod:energy");
+  n = data.num_atom_sum;
+  memory->create(data.position, 3*n, "fitpod:position");
+  memory->create(data.force, 3*n, "fitpod:force");
+  memory->create(data.atomtype, n, "fitpod:atomtype");
+
+  int nfiles = data.data_files.size(); // number of files
+  int nconfigs = 0;
+  int natoms = 0;
+  for (int i=0; i<nfiles; i++) {
+    read_exyz_file(&data.lattice[9*nconfigs], &data.stress[9*nconfigs], &data.energy[nconfigs],
+        &data.position[3*natoms], &data.force[3*natoms], &data.atomtype[natoms],
+        data.data_files[i], species);
+    nconfigs += data.num_config[i];
+    natoms += data.num_atom_each_file[i];
+  }
+
+  int len = data.num_atom.size();
+  data.num_atom_min = podptr->podArrayMin(&data.num_atom[0], len);
+  data.num_atom_max = podptr->podArrayMax(&data.num_atom[0], len);
+  data.num_atom_cumsum.resize(len+1);
+  podptr->podCumsum(&data.num_atom_cumsum[0], &data.num_atom[0], len+1);
+
+  data.num_config_cumsum.resize(nfiles+1);
+  podptr->podCumsum(&data.num_config_cumsum[0], &data.num_config[0], nfiles+1);
+
+  // convert all structures to triclinic system
+
+  int dim = 3;
+  double Qmat[dim*dim];
+  for (int ci=0; ci<len; ci++) {
+    int natom = data.num_atom[ci];
+    int natom_cumsum = data.num_atom_cumsum[ci];
+    double *x = &data.position[dim*natom_cumsum];
+    double *f = &data.force[dim*natom_cumsum];
+    double *lattice = &data.lattice[9*ci];
+    double *a1 = &lattice[0];
+    double *a2 = &lattice[3];
+    double *a3 = &lattice[6];
+
+    podptr->matrix33_inverse(Qmat, a1, a2, a3);
+    podptr->triclinic_lattice_conversion(a1, a2, a3, a1, a2, a3);
+    podptr->matrix33_multiplication(Qmat, lattice, Qmat, dim);
+    podptr->matrix33_multiplication(x, Qmat, x, natom);
+    podptr->matrix33_multiplication(f, Qmat, f, natom);
+  }
+
+  utils::logmesg(lmp, "minimum number of atoms: {}\n", data.num_atom_min);
+  utils::logmesg(lmp, "maximum number of atoms: {}\n", data.num_atom_max);
+}
+
+std::vector<int> CFITPOD::linspace(int start_in, int end_in, int num_in)
+{
+
+  std::vector<int> linspaced;
+
+  double start = static_cast<double>(start_in);
+  double end = static_cast<double>(end_in);
+  double num = static_cast<double>(num_in);
+
+  int elm;
+
+  if (num == 0) { return linspaced; }
+  if (num == 1)
+  {
+    elm = (int) std::round(start);
+    linspaced.push_back(elm);
+    return linspaced;
+  }
+
+  double delta = (end - start) / (num - 1);
+
+  for(int i=0; i < num-1; ++i)
+  {
+    elm = (int) std::round(start + delta * i);
+    linspaced.push_back(elm);
+  }
+
+  elm = (int) std::round(end);
+  linspaced.push_back(elm);
+
+  return linspaced;
+}
+
+std::vector<int> CFITPOD::shuffle(int start_in, int end_in, int num_in)
+{
+  int sz = end_in - start_in + 1;
+  std::vector<int> myvector(sz);
+
+  for (int i = 0; i<sz; i++)
+    myvector[i] = start_in + i;
+
+  unsigned seed = std::chrono::system_clock::now().time_since_epoch().count();
```

Comment:
Please use `platform::walltime()` multiplied by a suitable factor for increased portability and consistency.